osh/hub/scripts/osh-worker-manager

```diff
@@ -0,0 +1,180 @@
+#!/usr/bin/env python3
+# SPDX-License-Identifier: GPL-3.0-or-later
+# SPDX-FileCopyrightText: Copyright contributors to the OpenScanHub project.
+
+"""
+This script was originally written for the resalloc agent spawner daemon.
+But it may be used by any other resource allocation mechanism.
+This should be run as exactly the same user and group as the webserver. For example, `apache`.
+"""
+
+import logging
+import os
+import socket
+import tempfile
+
+logger = logging.getLogger(__name__)
+
+
+# On AWS, public hostname is accessible through this command on the vm:
+# `curl http://169.254.169.254/latest/meta-data/public-hostname`
+# TODO: Shall we use that instead of using ip address?
+def get_hostname_from_ip(ip_address):
+    logger.debug(f"Getting hostname for {ip_address}")
+    hostname = socket.gethostbyaddr(ip_address)[0]
+    return hostname
+
+
+def create_worker_in_kobo(worker_name):
+    from django.core.exceptions import ObjectDoesNotExist
+    from kobo.hub import models
+
+    # Check if a worker with this name already exists
+    try:
+        kobo_worker = models.Worker.objects.get(name=worker_name)
+    except ObjectDoesNotExist:
+        # If not, create a new one
+        kobo_worker = models.Worker.create_worker(worker_name)
+        kobo_worker.arches.add(models.Arch.objects.get(name="x86_64"))
```

Comment:
https://github.com/openscanhub/openscanhub/blob/55a0101e073122850ab84e0b43af76787c3fc96d/osh/hub/scripts/osh-worker-manager#L39

Reply:
Yes, I see that. But you have no guarantee that given architecture will exist in the database.  Sure, they are in the default fixtures but they are not loaded when you install the packages AFAIK.

Also, the second part of https://github.com/openscanhub/openscanhub/pull/234#discussion_r1519608321 about the `TODO:` was not addressed either.

Reply:
Would a worker that has `noarch` set, pick up a `x86_64` arch task?

Reply:
If the task has `x86_64` arch set, than no.  The current arch selection works like this: https://github.com/openscanhub/openscanhub/blob/f5e051c4337fed696481bcff11a4c48243cf2313/osh/hub/scan/scanner.py#L34-L38 
Therefore, if the `x86_64` architecture exists in the database, all tasks with `*-x86_64` mock profiles will use it, otherwise they will fall back on `noarch`.

Reply:
@siteshwar FYI,  https://github.com/openscanhub/openscanhub/pull/234#discussion_r1528764481 was written during review, so it is shown before https://github.com/openscanhub/openscanhub/pull/234#discussion_r1528765616 even though my was published alter.